### PR TITLE
First sector fail fix

### PR DESCRIPTION
**Pull request by jpbland1**
ZD 15415

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

File: src/update_flash.c [hunk 1]
```diff
@@ -492,6 +504,25 @@ static int RAMFUNCTION wolfBoot_update(int fallback_allowed)
                 wolfBoot_set_update_sector_flag(sector, flag);
         }
         sector++;
+        /* headers that can be in different positions depending on when the
```
Comment (dgarske) [inline src/update_flash.c:507]: Should this be wrapped in `#ifndef DISABLE_BACKUP`?

File: src/update_flash.c [hunk 2]
```diff
@@ -361,6 +361,19 @@ static int wolfBoot_delta_update(struct wolfBoot_image *boot,
...
 
+static int RAMFUNCTION wolfBoot_get_total_size(struct wolfBoot_image* boot,
+    struct wolfBoot_image* update)
+{
+    uint32_t total_size = 0;
+
+    /* Use biggest size for the swap */
+    total_size = boot->fw_size + IMAGE_HEADER_SIZE;
+    if ((update->fw_size + IMAGE_HEADER_SIZE) > total_size)
+            total_size = update->fw_size + IMAGE_HEADER_SIZE;
```
Comment (dgarske) [inline src/update_flash.c:372]: Please fix indent.

File: src/update_flash.c [hunk 3]
```diff
@@ -361,6 +361,19 @@ static int wolfBoot_delta_update(struct wolfBoot_image *boot,
     #define MAX_UPDATE_SIZE (size_t)((WOLFBOOT_PARTITION_SIZE - (2 *WOLFBOOT_SECTOR_SIZE)))
 #endif
 
+static int RAMFUNCTION wolfBoot_get_total_size(struct wolfBoot_image* boot,
```
Comment (dgarske) [inline src/update_flash.c:364]: Can we just make this inline and remove the RAMFUNCTION? `static inline int`?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/update_flash.c (hunk 1 above)
Comment (jpbland1) [reply src/update_flash.c:507]: it already is, that entire section of code is wrapped with that definition

**Review by dgarske — APPROVED** [review]

**Pull request merged**